### PR TITLE
MGMT-7925: Remove a redundant cluster registration event

### PR DESCRIPTION
**Pull request by nmagnezi**
# Assisted Pull Request

## Description

Upon cluster registration two events were produced:
- Registered cluster
- Successfully registered cluster

To match with host events, the 'Registered cluster' was removed.

Additionally, some registration failures did not produce any event.
This change addresses that problem as well.

## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by slaviered** [comment]
Just make sure that this message was not used by the dashboards

**Comment by nmagnezi** [comment]
> Just make sure that this message was not used by the dashboards

@michaellevy101 could you please check?

File: internal/cluster/cluster.go [hunk 1]
```diff
@@ -188,22 +188,15 @@ func NewManager(cfg Config, log logrus.FieldLogger, db *gorm.DB, eventsHandler e
...
 func (m *Manager) RegisterCluster(ctx context.Context, c *common.Cluster, v1Flag common.InfraEnvCreateFlag, v1ISOType models.ImageType) error {
-	err := m.registrationAPI.RegisterCluster(ctx, c, v1Flag, v1ISOType)
-	if err != nil {
-		eventgen.SendClusterRegistrationFailedEvent(ctx, m.eventsHandler, *c.ID, err.Error(), models.ClusterKindCluster)
-
-	} else {
-		eventgen.SendClusterRegisteredEvent(ctx, m.eventsHandler, *c.ID, models.ClusterKindCluster)
-	}
-	return err
+	return m.registrationAPI.RegisterCluster(ctx, c, v1Flag, v1ISOType)
```
Comment (masayag) [inline internal/cluster/cluster.go:191]: Is there a need to verify this won't change the behavior of automation test and their expectation?
Also, in the past we had a bug about this missing event: https://bugzilla.redhat.com/show_bug.cgi?id=2002735
Need to coordinate the change of behavior with them.

Comment (nmagnezi) [reply internal/cluster/cluster.go:191]: Based on a [slack discussion with QE](https://coreos.slack.com/archives/CUPJTHQ5P/p1638696331293400), they knowledge the impact on automation,[ but they are ok with it](https://coreos.slack.com/archives/CUPJTHQ5P/p1638722040301300?thread_ts=1638696331.293400&cid=CUPJTHQ5P).

**Review by avishayt — APPROVED** [review]

**Pull request merged**